trin-core/src/bin/generate-data.rs

```diff
@@ -0,0 +1,120 @@
+use discv5::enr::NodeId;
+use rand::{distributions::Alphanumeric, Rng};
+use sha3::{Digest, Sha3_256};
+use std::convert::TryInto;
+use std::process::Command;
+use std::sync::Arc;
+use structopt::StructOpt;
+use trin_core::portalnet::storage::{DistanceFunction, PortalStorage, PortalStorageConfig};
+use trin_core::portalnet::U256;
+use trin_core::utils::get_data_dir;
+
+pub fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let generator_config = GeneratorConfig::from_args();
+
+    let node_id = NodeId::random();
+
+    if generator_config.overwrite {
+        let _ = Command::new("rm")
+            .arg("-rf")
+            .arg(get_data_dir(node_id))
+            .output()
+            .expect("Failed to overwrite DB.");
+    }
+
+    let db = PortalStorage::setup_rocksdb(node_id)?;
+    let meta_db = PortalStorage::setup_sqlite(node_id)?;
+
+    let num_kilobytes = generator_config.kb;
+    let size_of_keys = 32;
+    let size_of_values = generator_config.value_size;
+
+    // For every 1 kb of data we store (key + value), RocksDB tends to grow by this many kb on disk...
+    // ...but this is a crude empirical estimation that works mainly with default value data size of 32
+    let data_overhead = 1.1783;
+    let num_entries = ((num_kilobytes * 1000) as f64 / (size_of_values) as f64) / data_overhead;
+    let num_of_entries = num_entries.round() as u32;
+
+    let storage_config = PortalStorageConfig {
+        // Arbitrarily set capacity at a quarter of what we're storing.
+        // todo: make this ratio configurable
+        storage_capacity_kb: (num_kilobytes / 4) as u64,
+        node_id,
+        distance_function: DistanceFunction::Xor,
+        db: Arc::new(db),
+        meta_db: Arc::new(meta_db),
+    };
+    let mut storage = PortalStorage::new(storage_config, |key| sha256(key))?;
```

Comment:
I don't have strong feelings either way, though I'm curious what your there reasoning is

Reply:
My intuition is that the configuration is for things that wer would like to tweak often, like different kind of parameters and values. The DB sounds more like a parameter to the PortalStorage instead of a configuration param, because probably we will not want to swap the db often.  Also when writing integration tests I would like to stub the db and directly pass it to the new function,  but it is personal preference I guess...